src/targets/registry.ts

```diff
@@ -396,78 +424,88 @@ export class RegistryTarget extends BaseTarget {
     return git;
   }
 
+  public async getValidItems(
+    version: string,
+    revision: string
+  ): Promise<RegistryConfig[]> {
+    return filterAsync(this.registryConfig, async registryConfig => {
+      if (!registryConfig.linkPrereleases && isPreviewRelease(version)) {
+        this.logger.info(
+          `Preview release detected, skipping ${registryConfig.canonicalName}`
+        );
+        return false;
+      }
+
+      // If we have onlyIfPresent specified, check that we have any of matched files
+      const onlyIfPresentPattern = registryConfig.onlyIfPresent;
+      if (onlyIfPresentPattern) {
+        const artifacts = await this.artifactProvider.filterArtifactsForRevision(
+          revision,
+          {
+            includeNames: onlyIfPresentPattern,
+          }
+        );
+        if (artifacts.length === 0) {
+          this.logger.warn(
+            `No files found that match "${onlyIfPresentPattern.toString()}", skipping the target.`
+          );
+          return false;
+        }
+      }
+      return true;
+    });
+  }
+
   /**
    * Modifies/adds meta information regarding the package we are publishing
    */
   public async publish(version: string, revision: string): Promise<any> {
-    if (!this.registryConfig.linkPrereleases && isPreviewRelease(version)) {
-      this.logger.info('Preview release detected, skipping the target');
-      return undefined;
-    }
+    const items = await this.getValidItems(version, revision);
 
-    // If we have onlyIfPresent specified, check that we have any of matched files
-    const onlyIfPresentPattern = this.registryConfig.onlyIfPresent;
-    if (onlyIfPresentPattern) {
-      const artifacts = await this.artifactProvider.filterArtifactsForRevision(
-        revision,
-        {
-          includeNames: onlyIfPresentPattern,
-        }
-      );
-      if (artifacts.length === 0) {
-        this.logger.warn(
-          `No files found that match "${onlyIfPresentPattern.toString()}", skipping the target.`
-        );
-        return undefined;
-      }
+    if (items.length === 0) {
+      this.logger.warn('No suitable items found, bailing');
+      return;
     }
 
-    return this.doPublish(version, revision);
-  }
+    await withTempDir(
+      async dir => {
+        const localRepo = {
+          dir,
+          git: await this.cloneRegistry(dir),
+        };
+        await Promise.all(
+          items.map(registryConfig =>
+            this.updateVersionInRegistry(
+              registryConfig,
+              localRepo,
+              version,
+              revision
+            )
+          )
+        );
```

Comment:
So this is the crucial part, where we update the filesystem for all registry sub-targets in parallel, and block on completion of all via `await Promise.all`. 👍

Reply:
Exactly 🎉